selection.cpp

```diff
@@ -18,16 +18,20 @@ Selection::~Selection()
 
 void Selection::init()
 {
-    QObject::connect(this, &Selection::loadQuiz, this, &Selection::quizLoading);
-    QObject::connect(this, &Selection::play, this, &Selection::playing);
-    QObject::connect(this, &Selection::finish, this, &Selection::print);
-
-    QSignalMapper *bMapper = new QSignalMapper;
-    connect(bMapper, SIGNAL(mappedInt(int)), this, SLOT(nextStep(int)));
-    connect(ui->pushButton_previous, SIGNAL(clicked(bool)), bMapper, SLOT(map()));
-    bMapper->setMapping(ui->pushButton_previous, PREVIOUS_QUESTION);
-    connect(ui->pushButton_next, SIGNAL(clicked(bool)), bMapper, SLOT(map()));
-    bMapper->setMapping(ui->pushButton_next, NEXT_QUESTION);
+    if(selectionIterator == 0)
+    {
+        QObject::connect(this, &Selection::loadQuiz, this, &Selection::quizLoading);
+        QObject::connect(this, &Selection::play, this, &Selection::playing);
+        QObject::connect(this, &Selection::finish, this, &Selection::print);
+
+        QSignalMapper *bMapper = new QSignalMapper;
+        connect(bMapper, SIGNAL(mappedInt(int)), this, SLOT(nextStep(int)));
+        connect(ui->pushButton_previous, SIGNAL(clicked(bool)), bMapper, SLOT(map()));
+        bMapper->setMapping(ui->pushButton_previous, PREVIOUS_QUESTION);
+        connect(ui->pushButton_next, SIGNAL(clicked(bool)), bMapper, SLOT(map()));
+        bMapper->setMapping(ui->pushButton_next, NEXT_QUESTION);
+    }
+    selectionIterator++;
```

Comment:
Na sličan način izbeći korišćenje promenljive ``selectionIterator`` kao što je to navedeno u ``Game`` i ``Highscore`` klasama.